bookkeeper-common/src/main/java/org/apache/bookkeeper/common/checksum/PureJavaCrc32C.java

```diff
@@ -0,0 +1,642 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.bookkeeper.common.checksum;
+
+import org.apache.bookkeeper.common.annotation.InterfaceAudience.Private;
+
+/**
+ * This class was taken from Hadoop: org.apache.hadoop.util.PureJavaCrc32C.
```

Comment:
We need to add hadoop to our NOTICE.

Reply:
And intel, since the code is copyrighted to them

Reply:
I understand you are into this NOTICE thing at this moment. but I don't see hadoop include this in their notice file.

If things get very complicated, I will prefer removing this class, rather than touching the NOTICE file.

Reply:
It's BSD, so it's in their LICENSE

https://github.com/apache/hadoop/blob/76e664e931bf0784620b69bc588bd51cf2a024e6/LICENSE.txt#L246

We don't actually need to put anything in NOTICE, since hadoop code is copyrighted to ASF.

Reply:
@sijie the discussion on legal from #888 also has a baring on this. So we either add the intel stuff to our LICENSE file(s), or don't take this code.